src/main/java/io/gravitee/reporter/common/utils/MetricsHelper.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.reporter.common.utils;
+
+import io.gravitee.reporter.api.v4.metric.AIMetrics;
+import io.gravitee.reporter.api.v4.metric.Metrics;
+import java.util.OptionalLong;
+import java.util.function.ToLongFunction;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class MetricsHelper {
+
+  public static OptionalLong aiInputTokens(@Nullable Metrics metrics) {
+    return aiTokens(metrics, AIMetrics::inputTokens);
+  }
+
+  public static OptionalLong aiOutputTokens(@Nullable Metrics metrics) {
+    return aiTokens(metrics, AIMetrics::outputTokens);
+  }
+
+  private static OptionalLong aiTokens(
+    @Nullable Metrics metrics,
+    ToLongFunction<AIMetrics> getter
+  ) {
+    return Stream
+      .ofNullable(metrics)
+      .flatMap(metrics1 -> Stream.ofNullable(metrics1.getAiMetrics()))
+      .mapToLong(getter)
+      .findAny();
+  }
+}
```

Comment:
what about adding "intelligent" getter on Metrics model 🤔 
It is not really Object oriented to write such helper no?